core/src/main/java/co/aikar/commands/CommandReplacements.java

```diff
@@ -83,7 +84,7 @@ public String replace(String text) {
         }
 
         // check for unregistered replacements
-        Pattern pattern = Pattern.compile("%.[^\\s]*");
+        Pattern pattern = Pattern.compile("%\\{.[^\\s]*}");
```

Comment:
lets move this to an acfpatterns static const (don't need to use the cache method here, just store result as const in that file and reference it.

though ideally need to move this find unregistered to be done at registration time instead of every single replacement :/

Reply:
> though ideally need to move this find unregistered to be done at registration time instead of every single replacement :/

Is this possible? At registration time we don't have access to the strings to be replaced.

Reply:
Well if instead of using ACFPatterns::getPattern, we store a replacementpattern cache map, and use .computeIfAbsent(), then only run the missing logic in the provider of the computeIfAbsent, then itll only run on the first time its asked for when the pattern is compiled and cached.

Reply:
So a <Pattern, String> map, where string is the value to be replaced.
When replacing, we loop through this map, matching the patterns and using the values.

And after that we seek for non-replaced placeholders in the String, if found, compile a Pattern for it and call computeIfAbsent using this pattern as key.
Inside the compute resolver, the unregistered replacements message is logged and the raw pattern is added as value.

@aikar is that what you have in mind?

Reply:
see final comment